bundles/org.eclipse.rap.rwt/src/org/eclipse/rap/rwt/internal/textsize/MeasurementOperator.java

```diff
@@ -72,12 +77,21 @@ Probe[] getProbes() {
   }
 
   void addProbeToMeasure( FontData fontData ) {
-    Probe probe = getApplicationContext().getProbeStore().getProbe( fontData );
+    ProbeStore probeStore = getProbeStore();
+    Probe probe = probeStore.getProbe( fontData );
     if( probe == null ) {
-      probe = getApplicationContext().getProbeStore().createProbe( fontData );
+      probe = probeStore.createProbe( fontData );
     }
     probes.add( probe );
   }
+  
+  ProbeStore getProbeStore() {
+    if(RWTProperties.isTextSizeStoreSessionScoped()) {
```

Comment:
To avoid performance problems by querying a system property too often, please keep it in a static final boolean field.